### ISSUE TITLE
Delete encounters from frontend

### ISSUE THREAD

**Issue by noahjade** [se701team3/Forgettable#46]
## Description
If users have accidentally created an encounter, or don't wish to keep an old one, we need to allow them to delete it. This will be done on the People + encounters page, by clicking a button on the relevant encounters card.

The following design is outdated (edit is out of scope for MVP), but the position of the edit button will be where the delete button is. Make sure to refer to current UI designs when implementing to be up to date.

![image](https://user-images.githubusercontent.com/28725774/156984119-7d41f709-9366-4c16-89da-3bf8752884a1.png)

## Acceptance Criteria

- [ ] When the user clicks delete, a modal pops up prompting the user to confirm their decision.
- [ ] After the user has confirmed their decision, the API is called and the encounter is deleted from the database.
- [ ] When the encounter is deleted, the list of encounter cards updates to reflect this change.

## Implementation details

- This is dependent on #45 and the backend implementing a delete encounters API endpoint.

## Testing notes

- Ensure the list updates to reflect the change.
- Make sure if the user cancels out of the modal, the encounter is not deleted.

## Reviewer
@uoa-lchu090
